homeassistant/components/vizio/config_flow.py

```diff
@@ -99,6 +103,23 @@ def __init__(self) -> None:
                     errors["base"] = "tv_needs_token"
 
             if not errors:
+                unique_id = await VizioAsync.get_unique_id(
+                    user_input[CONF_HOST],
+                    user_input.get(CONF_ACCESS_TOKEN),
+                    user_input[CONF_DEVICE_CLASS],
+                )
+
+                # Abort flow if flow in progress or component with same unique ID matches new config entry
+                try:
+                    if unique_id and await self.async_set_unique_id(
+                        unique_id=unique_id, raise_on_progress=True
+                    ):
+                        return self.async_abort(reason="already_setup")
+                except AbortFlow:
```

Comment:
We don't need to catch this. It's caught by the flow manager, which makes sure to create the correct result.